pkg/storage/snapshot/snapshot.go

```diff
@@ -707,6 +708,11 @@ func (ctrl *VMSnapshotController) updateSnapshotStatus(vmSnapshot *snapshotv1.Vi
 		updateSnapshotCondition(vmSnapshotCpy, newProgressingCondition(corev1.ConditionFalse, "Operation complete"))
 		updateSnapshotCondition(vmSnapshotCpy, newReadyCondition(corev1.ConditionTrue, "Operation complete"))
 		updateSnapshotSnapshotableVolumes(vmSnapshotCpy, content)
+		// Update "kubevirt_vmsnapshot_timestamp_seconds" metric if the snapshot succeeded
+		if vmSnapshot.Status != nil && vmSnapshot.Status.Phase == snapshotv1.Succeeded {
+			virt_controller.NewVMSnapshotCreated(vmSnapshot, content)
+		}
```

Comment:
1. You are repeating the `snapshot.Status != nil` here and inside `NewVMSnapshotCreated`
2. Can we add a phase label to the metric, and also track non `Succeeded` snapshots?
3. If we really only want to only track `Succeeded`, can we move that validation inside `NewVMSnapshotCreated`? I think it would look better here

Reply:
we agreed to add only snapshot that Succeeded with @sradco @enp0s3 on our sig meeting, you think we should also add failed snapshots?

Reply:
and @machadovilaca I moved the logic as requested in 3.

Reply:
I think is make sense to report this to Failed and and aborted snapshots, but as additional metrics.

Reply:
Snapshots are not automatically deleted in any case either succeeded nor failed. @avlitman Regarding if we should add a metric for failed snapshot creation Im not sure, possibly, I think its a question more relevant maybe to UX as what information interests the user.

Reply:
@avlitman due to what @ShellyKa13 mentioned, please update the metric name to  `kubevirt_vmsnapshot_succeeded_timestamp_seconds`, since we will be adding metrics also for the other snapshots statuses. It is needed in order to identify snapshots that can be deleted and free up space.